### ISSUE TITLE
Sourcing multiple files with FileParser

### ISSUE THREAD

**Issue by dorkster** [clintbellanger/flare-engine#1392]
For some of our game data files, keeping things organized can get out of control fast. Examples that come to mind are `items/items.txt` and `powers/powers.txt`, which both have a lot of content.

It would be nice to be able to split these up as we see fit, and simply reference them from the main file. We already have the **APPEND** directive, which allows us to combine files of the same name across mods. I propose an **INCLUDE** directive, which allows us to combine files of *different* names. So our `items/items.txt` file might end up looking like this:

```
INCLUDE items/armor.txt
INCLUDE items/weapons.txt
INCLUDE items/consumables.txt
# etc...
```

**Comment by igorko** [comment]
Looks interesting, it will be easier to edit and better organized.

**Comment by pennomi** [comment]
Agreed. This should make things MUCH easier to maintain.

**Comment by dorkster** [comment]
This is now implemented here https://github.com/clintbellanger/flare-engine/commit/ca47cd17b6fc48e2452f964d49f1b005771dbe9e

@igorko I also made an interesting discovery. During the process, I was also able to fix the bug where the first line of text files was eaten. This seemed to occur more on Android, but it showed up on the desktop build when I created a file that only had an INCLUDE directive.

As it turns out, `getline()` (sometimes?) destroys the line it is reading. This is bad since we want to parse the beginning portion of the file twice: once when looking for APPEND and again when doing normal parsing. The clearest solution was to simply re-open the file after first pass, restoring any lines that were destroyed.

I have tested on Android with my changes and unaltered game data. Everything worked as expected.
